src/components/views/right_panel/RoomSummaryCard.tsx

```diff
@@ -287,7 +287,7 @@ const RoomSummaryCard: React.FC<IProps> = ({ room, permalinkCreator, onClose })
 
     const onRoomPollHistoryClick = (): void => {
         Modal.createDialog(PollHistoryDialog, {
-            roomId: room.roomId,
+            room,
             matrixClient: cli,
             permalinkCreator,
```

Comment:
This looks to me like it ought to introduce a type-checking error, and I'm slightly mystified as to why it doesn't. Am I missing some reason why we can rely on `permalinkCreator` to be defined despite the `?` in `IProps` ?

Reply:
(it feels like the right solution here is to make `permalinkCreator` a required prop for `RoomSummaryCard` and add a `!` in RightPanel. YMMV though)

Reply:
`createDialog` has very soft types

Reply:
> `createDialog` has very soft types

Ah that explains why typescript isn't catching it. Still feel like it's a thing that we should address though.

To expand on my thinking here: AIUI currently we're assuming that whenever RightPanel is given a Room, it is also given a permalink creator. That's okayyyy though I wish that assumption was made explicit either via comments or, better, via the typing system. Still it's not a new situation so doesn't particularly need fixing in this PR. What we *can* do in this PR is limit the impact of that assumption by sticking a `!` in RightPanel rather than softening the types for RoomSummaryCard.